owid/datautils/io/df.py

```diff
@@ -0,0 +1,84 @@
+"""DataFrame io operations."""
+import pandas as pd
+from pathlib import Path
+from owid.datautils.decorators import enable_file_download
+from typing import Any, Optional, Union, List
+
+
+COMPRESSION_SUPPORTED = ["gz", "bz2", "zip", "xz", "zst", "tar"]
+
+
+@enable_file_download("file_path")
+def from_file(
+    file_path: Union[str, Path], file_type: Optional[str] = None, **kwargs: Any
+) -> Union[pd.DataFrame, List[pd.DataFrame]]:
+    """Read a CSV or excel file.
```

Comment:
The docstring needs to be updated: `Read data from a file.`. Also, the examples need to be rephrased: `For example: "file.csv" will be read as a csv file, and "http://my/domain/file.xlsx" will be read as an excel file.`